pkg/sql/opt/xform/join_funcs.go

```diff
@@ -1818,3 +1818,40 @@ func (c *CustomFuncs) splitValues(
 	}
 	return localVals, remoteVals
 }
+
+// getfilteredCanonicalScan looks at a *ScanExpr or *SelectExpr "relation" and
+// returns the input *ScanExpr and FiltersExpr, along with ok=true, if the Scan
+// is a canonical scan. If "relation" is a different type, or if it's a
+// *SelectExpr with an Input other than a *ScanExpr, ok=false is returned. Scans
+// or Selects with no filters may return filters as nil.
+func (c *CustomFuncs) getfilteredCanonicalScan(
```

Comment:
Just noticed it should be `getFiltered...` not `getfiltered...`. No need to change this for the backport since that's how it's spelled on master, but you can fix on master in a future PR if you'd like.

Reply:
> Just noticed it should be `getFiltered...` not `getfiltered...`. No need to change this for the backport since that's how it's spelled on master, but you can fix on master in a future PR if you'd like.

OK, thanks.